blog/_posts/2016-10-14-dotty-errors.md

```diff
@@ -0,0 +1,103 @@
+---
+layout: blog
+post-type: blog
+by: Felix Mulder
+title: "Awesome Error Messages for Dotty"
+---
+
+One thing that really excites me about being part of the core group of
+developers working on [Dotty](http://lampepfl.github.io/dotty) is my chance to
+impact usability. A lot of thought has gone into designing Dotty to be as fast
+and structurally sound as possible.  Now comes the next step - adding a new
+level of usability for the compiler and the surrounding tools.
+
+We've looked at how other modern languages like
+[Elm](http://elm-lang.org/blog/compiler-errors-for-humans) and
+[Rust](https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html)
+handle compiler warnings and error messages, and come to realize that Dotty is
+actually in great shape to provide comprehensive and easy to understand error
+messages in the same spirit.
+
+Let's dive right into some examples, let's say you have this code:
+
+    try {
+      1 + 2
+    }
+
+It doesn't really make sense to put this in a `try`-block for two reasons:
+
+1. It doesn't throw an exception
+2. It doesn't have a `catch` or `finally` clause
+
+So let's say we compile this file using scalac, we get something like:
+
+<pre>
+test.scala:2: warning: A try without a catch or finally is equivalent to putting its body in a block; no exceptions are handled.
+     try {
+       ^
+       one warning found
+</pre>
+
+This is helpful, but it has a couple of drawbacks:
+
+1. If we have a bazillion errors, it will be hard to read
```

Comment:
Yes - the new errors have a sane division between errors so you can see where one ends and another begins.